src/buffer.rs

```diff
@@ -35,60 +42,126 @@ impl AudioBuffer {
+    pub fn copy_from_channel_with_offset(
+        &self,
+        destination: &mut Vec<f32>,
+        channel_number: usize,
+        offset: usize,
+    ) {
+        // [spec] Let buffer be the AudioBuffer with 𝑁𝑏 frames, let 𝑁𝑓 be the number
+        // of elements in the destination array, and 𝑘 be the value of bufferOffset.
+        // Then the number of frames copied from buffer to destination is max(0,min(𝑁𝑏−𝑘,𝑁𝑓)).
+        // If this is less than 𝑁𝑓, then the remaining elements of destination are not modified.
+        //
+        // we use the `capacity` instead of `len` because destination is a write
+        // buffer and we don't care of its current values (cf. `copy_to_channel`)
+        let dest_capacity = destination.capacity();
+        let max_frame = (self.length() - offset).min(dest_capacity).max(0);
+        let channel = self.channel_data(channel_number).as_slice();
+
+        for index in 0..max_frame {
+            if index < destination.len() {
+                destination[index] = channel[index + offset];
+            } else {
+                destination.push(channel[index + offset]);
+            }
+        }
+    }
+
+    /// Copy data from a given source to the given channel.
+    pub fn copy_to_channel(&mut self, source: &[f32], channel_number: usize) {
+        self.copy_to_channel_with_offset(source, channel_number, 0);
+    }
+
+    /// Copy data from a given source to the given channel starting at `offset`.
+    pub fn copy_to_channel_with_offset(
+        &mut self,
+        source: &[f32],
+        channel_number: usize,
+        offset: usize,
+    ) {
+        // [spec] Let buffer be the AudioBuffer with 𝑁𝑏 frames, let 𝑁𝑓 be the number
+        // of elements in the source array, and 𝑘 be the value of bufferOffset. Then
+        // the number of frames copied from source to the buffer is max(0,min(𝑁𝑏−𝑘,𝑁𝑓)).
+        // If this is less than 𝑁𝑓, then the remaining elements of buffer are not modified.
+        //
+        // we use the `len` instead of `capacity` because source is a read buffer and we
+        // want to be sure we don't access some undefined index (cf. `copy_from_channel`)
+        let src_len = source.len();
+        let max_frame = (self.length() - offset).min(src_len).max(0);
+        let channel = self.channel_data_mut(channel_number).as_mut_slice();
+
+        channel[offset..(max_frame + offset)].copy_from_slice(&source[..max_frame]);
+    }
+
+    /// Return a read-only copy of the underlying data of the channel
+    pub fn get_channel_data(&self, channel_number: usize) -> &[f32] {
+        // [spec] According to the rules described in acquire the content either allow writing
+        // into or getting a copy of the bytes stored in [[internal data]] in a new Float32Array
+        self.channel_data(channel_number).as_slice()
     }
 
     /// Channel data as slice
-    pub fn channels(&self) -> &[ChannelData] {
+    pub(crate) fn channels(&self) -> &[ChannelData] {
```

Comment:
Marking these function `pub(crate)` is good for now, as they are not part of the spec